### PR TITLE
Implement Project config endpoints

### PR DESCRIPTION
**Pull request by PMooreWand**
Adding implementation of project/config.
Get and Put commands implemented.
Added new parser method to ProjectsParser.
Added new tests to ProjectParserTest and
ProjectApiClientTest

### CODE REVIEW

**Comment by PMooreWand** [comment]
Maybe an api mismatch here running against 3.7.5 version get issue on mapping extensionPanelNames
`
    java.lang.IllegalArgumentException: Can not set com.google.common.collect.ImmutableMap field com.google.gerrit.extensions.api.projects.ConfigInfo.extensionPanelNames to com.google.gson.internal.LinkedTreeMap`

**Comment by PMooreWand** [comment]
Closing pull for now trusted example in gerrit docs but gson parsing fails to map the extensionspanel map from json. Closing until get that resolved

**Comment by uwolfer** [comment]
@PMooreWand Does it fail in Gerrit itself? Also using a curl command? Then you should report it against Gerrit IMO.

**Comment by PMooreWand** [comment]
It's Gson deserialisation the ConfigInfo between 2.16 and 3.7 changed from
` public Map<String, List<String>> extensionPanelNames;`
to
`public ImmutableMap<String, ImmutableList<String>> extensionPanelNames;` 
So probably more more the case that I can not just call gson.fromJson without more logic for the mapping.

**Comment by uwolfer** [comment]
> So probably more the case that I can not just call gson.fromJson without more logic for the mapping.

Not sure, but probably this helps in your case? https://stackoverflow.com/questions/13623175/deserialize-to-an-immutablemap-with-gson

File: src/test/resources/com/urswolfer/gerrit/client/rest/http/projects/configInfo.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        "gc": {
+            "method": "POST",
+            "label": "Run GC",
+            "title": "Triggers the Git Garbage Collection for this project.",
+            "enabled": true
+            }
+        },
+    "commentlinks": {},
+    "extension_panel_names":{
+           "lfs": ["sadasd","asdsada"]
```
Comment (PMooreWand) [inline src/test/resources/com/urswolfer/gerrit/client/rest/http/projects/configInfo.json:61]: Worth noting here that I couldn't find any examples of what this field is meant to hold, It isn't in the documentation, so this is just to make a String,List<string> entry

**Comment by uwolfer** [comment]
Happy to see that you were able to resolve the issue. Could you please reformat the newly created code (there are some minor code style inconsistencies mostly related to white spaces)?

**Pull request merged**